crates/runtime/src/dataconnector/mysql.rs

```diff
@@ -186,4 +191,205 @@ impl DataConnector for MySQL {
             }
         }
     }
+
+    fn metrics_provider(&self) -> Option<Arc<dyn MetricsProvider>> {
+        Some(Arc::new(MySQLMetricsProvider::new(
+            self.mysql_factory.conn_pool_metrics(),
```

Comment:
This is where we get the metrics from the MySQL connection pool, see https://github.com/blackbeam/mysql_async/blob/master/src/conn/pool/metrics.rs for the underlying metrics that are tracked.

I quite like the approach this library took for metrics - it pairs nicely with the AsyncInstruments from OTel. Because each metric is an Atomic, we can get a reference to the struct (via Arc<Metrics>) and whenever we want to read the metric value, we just load it from the Atomic.